src/preact/base-element.js

```diff
@@ -142,34 +143,46 @@ const childIdGenerator = sequentialIdGenerator();
  * @template API_TYPE
  */
 export class PreactBaseElement extends AMP.BaseElement {
-  /**
-   * @return {boolean}
-   * @nocollapse
-   */
+  /** @override @nocollapse */
+  static V1() {
+    return true;
+  }
+
+  /** @override @nocollapse */
   static requiresShadowDom() {
     // eslint-disable-next-line local/no-static-this
     return usesShadowDom(this);
   }
 
+  /** @override @nocollapse */
+  static prerenderAllowed() {
+    // eslint-disable-next-line local/no-static-this
+    const Ctor = this;
+    return !Ctor['loadable'];
```

Comment:
Remind me please, the `layoutCallback` won't be called until `DisplayObserver` says it's displayable?

Reply:
The `layoutCallback` is not called in V1 at all since it's just a part of a single "build" phase. However, an element can (a) either provide a different `static prerenderAllowed() {}`, or (b) delay some parts of it itself via some observer. Here, I just thought it was a good default - if an element is known not to load anything at all, it's probably safe to prerender it.

Reply:
It's reverse: if there's loadable then preloading will definitely not happen. I just thought it was a good default. If there's nothing to load - then prerendering seems like would be safe in most of cases. But if you think it's too risky - we can remove and let components add it explicitly. LMK.